### ISSUE TITLE
(Optionally) use BCC instead of TO in mailer.NewMessageFrom()

### ISSUE THREAD

**Issue by ltGuillaume** [gogs/gogs#5681]
I noticed that after creating a pull request, I was notified about the conversation and merging via email messages that included all other watchers in the field "To".

One could argue that those email addresses are made public when pushing commits anyway, but it’s still a whole different thing to actually have people get our email addresses in their inboxes directly (we can't assume everyone has their affairs in order security-wise).

Hence I would propose the substitution of `msg.SetHeader("To", to...)` with `msg.SetHeader("Bcc", to...)` in `mailer.NewMessageFrom()` in order to circumvent the risk desribed above.

**Comment by achilleas-k** [comment]
We made this change on a fork we host for the same reasons.  Unfortunately, having an email sent without To address and only BCC made it likely to be marked as spam in a lot of situations.  A lot of users were having their activation emails lost.  In this case it was a university email filter that would just bounce the email back, so it didn't even end up in users spam folders.

I'm curious if an alternative approach might be to send one message per user instead, with just their address in the To field?

**Comment by unknwon** [comment]
>I'm curious if a better alternative approach might be to send one message per user instead, with just their address in the To field? Wondering if this might cause any potential issues.

@achilleas-k I think that's might too costly?

We can add an config option to `[mailer]` section and user can configure to use BCC instead of To.

**Comment by ltGuillaume** [comment]
@unknwon I would argue that this should be the default behavior, as it is bad practice to disclose email addresses to others like this. It would not surprise me at all if this is in violation of the GDPR.

**Comment by unknwon** [comment]
@ltGuillaume

>I would argue that BCC'ing should be the default behavior, as it is bad practice to disclose email addresses to others like this. It would not surprise me at all if this is in violation of the GDPR.

I do not think it is a good idea to change the default behavior, people rely on things that work in the way those are intended or not, as long as they are not deal breakers.

**Comment by achilleas-k** [comment]
> > I'm curious if a better alternative approach might be to send one message per user instead, with just their address in the To field? Wondering if this might cause any potential issues.
> 
> @achilleas-k I think that's might too costly?

I had the same concern, but for our user volume it's not an issue.  I understand it might be if 100 people are subscribed to an issue, for instance.

Perhaps the better solution is to BCC recipients and add a "noreply" or dummy address in the `To:` field since, as I understand it, the real issue that's triggering spam detection is the lack of a recipient address.
